File: app/src/main/java/com/cbr/behancesampleapp/dagger/scope/FragmentScope.java
package com.cbr.behancesampleapp.dagger.scope;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import javax.inject.Scope;

/**
 * Created by dimitrios on 24/08/2017.
 * <p>
 * Taken From https://github.com/googlesamples/android-architecture/blob/todo-mvp-dagger/
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface FragmentScope {

}

File: app/src/main/java/com/cbr/behancesampleapp/domain/dagger/scope/FragmentScope.kt
package com.cbr.behancesampleapp.domain.dagger.scope

import javax.inject.Scope

/**
 * Created by dimitrios on 24/08/2017.
 *
 *
 * Taken From https://github.com/googlesamples/android-architecture/blob/todo-mvp-dagger/
 */
@Scope
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class FragmentScope
